### PR TITLE
Retarget to .NET Standard

### PR DESCRIPTION
**Pull request by mjrousos**
This re-targets all projects except for test, CLI, and MSBuild projects to netstandard2.0. At the moment, net5.0 is fine, but if the tool can target netstandard2.0 that may open up more hosting possibilities in the future.

### CODE REVIEW

File: src/components/Microsoft.DotNet.UpgradeAssistant.Portability/Service/ApiInformation.cs [hunk 1]
```diff
@@ -27,12 +27,12 @@ public bool IsSupported()
...
             foreach (var supported in Supported)
             {
-                if (supported.Identifier.Contains("Standard", StringComparison.OrdinalIgnoreCase))
+                if (supported.Identifier.ToUpperInvariant().Contains("STANDARD"))
                 {
                     return true;
                 }
 
-                if (supported.Identifier.Contains("Core", StringComparison.OrdinalIgnoreCase))
+                if (supported.Identifier.ToUpperInvariant().Contains("CORE"))
```
Comment (sunandabalu) [inline src/components/Microsoft.DotNet.UpgradeAssistant.Portability/Service/ApiInformation.cs:35]: How is this better than StringComparison.OrdinalIgnoreCase?

**Comment by sunandabalu** [comment]
Looks like the CI is failing with the move to netstandard

File: src/components/Microsoft.DotNet.UpgradeAssistant.Portability/Service/ApiInformation.cs (hunk 1 above)
Comment (mjrousos) [reply src/components/Microsoft.DotNet.UpgradeAssistant.Portability/Service/ApiInformation.cs:35]: It's not, but it works on .NET Standard :)

File: src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Templates/TemplateInserterStep.cs [hunk 2]
```diff
@@ -259,7 +259,7 @@ private bool ItemMatches(string filePath, ItemSpec expectedItem)
             if (expectedItem.Keywords.Any())
             {
                 var fileContents = File.ReadAllText(filePath);
-                if (expectedItem.Keywords.Any(k => !fileContents.Contains(k, StringComparison.Ordinal)))
+                if (expectedItem.Keywords.Any(k => !fileContents.Contains(k)))
```
Comment (KSchlobohm) [inline src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Templates/TemplateInserterStep.cs:262]: is this intended as a case-sensitive search?

Comment (mjrousos) [reply src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Templates/TemplateInserterStep.cs:262]: Yes, it's intended to be case-sensitive.

**Comment by sunandabalu** [comment]
> Looks like the CI is failing with the move to netstandard

The tests were failing coz its expecting the dotnet version to be net5,0? When arcade update tried to change the global.json to 6.0-preview, the tests failed in the same way #304 , are the tests expecting a specific version?

**Comment by KSchlobohm** [comment]
We need to review the build error from the devOps pipeline to merge this PR. When running through VS 2019 all tests are passing for this branch.

'''
Found conflicts between different versions of "Microsoft.Bcl.AsyncInterfaces" that could not be resolved.  These reference conflicts are listed in the build log when log verbosity is set to detailed.
'''

Also beginning to note drift between this branch and 'main' due to other branches being merged.

**Comment by mjrousos** [comment]
Yea, I've been slow to merge this one because it's not a high priority. I'll update it and take a look at the build issues soon.

**Comment by mjrousos** [comment]
I've fixed the `Microsoft.Bcl.AsyncInterfaces` error by updating Roslyn libraries to 3.9.0 (3.8.0 had a weird `Microsoft.Bcl.AsyncInterfaces` reference that was causing problems). Hopefully tests pass now (they pass locally). We will see.

**Comment by mjrousos** [comment]
@sunandabalu @KSchlobohm - Looks like tests are passing now. If you could take one more quick look and approve it things look ok, that would be great!

**Review by KSchlobohm — APPROVED** [review]
Looks good!

**Comment by Saibamen** [comment]
>but if the tool can target netstandard2.0 that may open up more hosting possibilities in the future.

Which possibilities?

**Comment by mjrousos** [comment]
@Saibamen, currently, Upgrade-Assistant runs as a .NET 5 console app. If most of the app targets .NET Standard, then it will be easier to use the tool from within hosts running on .NET Framework (like Visual Studio, for example).

**Pull request merged**